### PR TITLE
Add `_.propertyDeep` for #1060

### PR DESCRIPTION
**Pull request by octref**
Lemme know if I did anything wrong!

### CODE REVIEW

**Comment by octref** [comment]
Reference #1060 , so people can find this PR from there.

File: lodash.src.js [hunk 1]
```diff
@@ -11204,6 +11226,34 @@
...
+     * var getAge  = _.propertyDeep(['user', 'age']);
+     *
+     * _.map(users, getName);
+     * // => ['fred', 'barney']
+     *
+     * _.map(users, getAge);
+     * // => [undefined, 36]
+     */
+    function propertyDeep(keys) {
+      keys = baseMap(keys, function(key) { return key + '' });
```
Comment (megawac) [inline lodash.src.js]: you can do baseMap(String) here

File: lodash.src.js [hunk 2]
```diff
@@ -2599,6 +2599,28 @@
...
+     * @param {string[]} keys Keys that specify the property to get.
+     * @returns {Function} Returns the new function.
+     */
+    function basePropertyDeep(keys) {
+      return function(object) {
+        var index = -1,
+            length = keys.length;
+
+        while (++index < length) {
+          object = baseProperty(keys[index])(object);
```
Comment (megawac) [inline lodash.src.js]: what's the reason to use baseProp here instead of a simple accessor

File: lodash.src.js (hunk 1 above)
Comment (octref) [inline lodash.src.js]: @megawac Nice, thanks!

Comment (octref) [inline lodash.src.js]: But I guess maybe `baseToString` is a better option?

Comment (megawac) [inline lodash.src.js]: Yeah, probs

Comment (jdalton) [inline lodash.src.js]: `arrayMap`.

Comment (octref) [inline lodash.src.js]: `arrayMap` doesn't allow falsey first argument though. Should I make `propertyDeep` rejectFalsey?

File: lodash.src.js (hunk 2 above)
Comment (octref) [inline lodash.src.js]: Right, my bad.

**Comment by jdalton** [comment]
Can you add `propertyDeepOf` too.

**Comment by octref** [comment]
Sure thing, give me an hour.

**Pull request merged**